### PR TITLE
Add new utility methods for GraalVM and refactor

### PR DESCRIPTION
**Pull request by ilopmar**
See https://github.com/micronaut-projects/micronaut-sql/issues/275

Once this is merged and Micronaut 2.0.2 released I will merge https://github.com/micronaut-projects/micronaut-sql/pull/278 with the changes.

### CODE REVIEW

File: core/src/main/java/io/micronaut/core/graal/AutomaticFeatureUtils.java [hunk 1]
```diff
@@ -44,67 +45,118 @@
...
      * Marks the given class to be initialized at build time, only if it is present.
      *
-     * @param access the {@link BeforeAnalysisAccess} instance
-     * @param className the class name
+     * @param access     The {@link BeforeAnalysisAccess} instance
+     * @param classNames The class names
      */
-    public static void initializeAtBuildTime(BeforeAnalysisAccess access, String className) {
-        findClass(access, className).ifPresent(RuntimeClassInitialization::initializeAtBuildTime);
+    public static void initializeAtBuildTime(BeforeAnalysisAccess access, String... classNames) {
```
Comment (sdelamo) [inline core/src/main/java/io/micronaut/core/graal/AutomaticFeatureUtils.java:51]: Even if the class is flagged as internal. Should not we keep the method and annotate it with `@Deprecated` and in the java doc tell to use the new one: 

```java
    public static void initializeAtBuildTime(BeforeAnalysisAccess access, String className) {
```

From the old one we can the new one.

File: core/src/main/java/io/micronaut/core/graal/AutomaticFeatureUtils.java [hunk 2]
```diff
@@ -44,67 +45,118 @@
...
 
     /**
      * Marks the given class to be initialized at runtime, only if it is present.
      *
-     * @param access the {@link BeforeAnalysisAccess} instance
-     * @param className the class name
+     * @param access     The {@link BeforeAnalysisAccess} instance
+     * @param classNames The class names
      */
-    public static void initializeAtRunTime(BeforeAnalysisAccess access, String className) {
```
Comment (sdelamo) [inline core/src/main/java/io/micronaut/core/graal/AutomaticFeatureUtils.java:60]: Even if the class is flagged as internal. Should not we keep the method and annotate it with `@Deprecated` and in the java doc tell to use the new one: 

```java
    public static void initializeAtBuildTime(BeforeAnalysisAccess access, String className) {
```

From the old one we can the new one.

File: core/src/main/java/io/micronaut/core/graal/AutomaticFeatureUtils.java (hunk 1 above)
Comment (ilopmar) [reply core/src/main/java/io/micronaut/core/graal/AutomaticFeatureUtils.java:51]: No, we don't. I asked about this at one engineering meeting and it's totally ok to do "whatever" we want with `@Internal` classes. Besides that, both signatures are compatible.

File: core/src/main/java/io/micronaut/core/graal/AutomaticFeatureUtils.java (hunk 2 above)
Comment (ilopmar) [reply core/src/main/java/io/micronaut/core/graal/AutomaticFeatureUtils.java:60]: Same as before.

**Pull request merged**